Rust compilation when building slim.Dockerfile fails

`docker build -t live-compositor -f build_tools/docker/slim.Dockerfile .` on current master (`65c443cfff7773c37dab632f4f33d60c125ac476` ) fails with:

Similar to https://github.com/software-mansion/live-compositor/issues/656

```
~/Documents/live-compositor docker build -t live-compositor -f build_tools/docker/slim.Dockerfile .
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon  56.55MB
Step 1/26 : FROM ubuntu:noble-20240423 as builder
 ---> 07db2ffa22da
Step 2/26 : SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ---> Using cache
 ---> d0e80dddec21
Step 3/26 : ARG USERNAME=compositor
 ---> Using cache
 ---> 1d325bce746a
Step 4/26 : ARG RUST_VERSION=1.74
 ---> Using cache
 ---> 170a779f341e
Step 5/26 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> d44f8646790c
Step 6/26 : RUN apt-get update -y -qq &&   apt-get install -y     build-essential curl pkg-config libssl-dev libclang-dev git sudo     libegl1-mesa-dev libgl1-mesa-dri libxcb-xfixes0-dev mesa-vulkan-drivers     ffmpeg libavcodec-dev libavformat-dev libavfilter-dev libavdevice-dev libopus-dev &&   rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> b94d50ae7344
Step 7/26 : RUN curl https://sh.rustup.rs/ -sSf | bash -s -- -y
 ---> Running in 76b6de7a015b
info: downloading installer
info: profile set to 'default'
info: default host triple is aarch64-unknown-linux-gnu
info: syncing channel updates for 'stable-aarch64-unknown-linux-gnu'
info: latest update on 2024-09-05, rust version 1.81.0 (eeb90cda1 2024-09-04)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'

info: default toolchain set to 'stable-aarch64-unknown-linux-gnu'
  stable-aarch64-unknown-linux-gnu installed - rustc 1.81.0 (eeb90cda1 2024-09-04)


Rust is installed now. Great!

To get started you may need to restart your current shell.
This would reload your PATH environment variable to include
Cargo's bin directory ($HOME/.cargo/bin).

To configure your current shell, you need to source
the corresponding env file under $HOME/.cargo.

This is usually done by running one of the following (note the leading DOT):
. "$HOME/.cargo/env"            # For sh/bash/zsh/ash/dash/pdksh
source "$HOME/.cargo/env.fish"  # For fish
 ---> Removed intermediate container 76b6de7a015b
 ---> 5076b086e5e3
Step 8/26 : RUN source ~/.cargo/env && rustup install $RUST_VERSION && rustup default $RUST_VERSION
 ---> Running in 451fac50c450
info: syncing channel updates for '1.74-aarch64-unknown-linux-gnu'
info: latest update on 2023-12-07, rust version 1.74.1 (a28077b28 2023-12-04)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'

  1.74-aarch64-unknown-linux-gnu installed - rustc 1.74.1 (a28077b28 2023-12-04)

info: checking for self-update
info: using existing install for '1.74-aarch64-unknown-linux-gnu'
info: default toolchain set to '1.74-aarch64-unknown-linux-gnu'

  1.74-aarch64-unknown-linux-gnu unchanged - rustc 1.74.1 (a28077b28 2023-12-04)

 ---> Removed intermediate container 451fac50c450
 ---> 5e4021ea1e48
Step 9/26 : COPY . /root/project
 ---> 0cd2ee7db48a
Step 10/26 : WORKDIR /root/project
 ---> Running in 2231170c1972
 ---> Removed intermediate container 2231170c1972
 ---> 6e3b39fab118
Step 11/26 : RUN source ~/.cargo/env && cargo build --release --no-default-features
 ---> Running in 674a715cc487
warning: /root/project/compositor_web/Cargo.toml: unused manifest key: build
    Updating crates.io index
    Updating git repository `https://github.com/grovesNL/glyphon`
    Updating git repository `https://github.com/membraneframework-labs/mp4-rust.git`
    Updating git repository `https://github.com/membraneframework-labs/schemars`
 Downloading crates ...
  Downloaded foreign-types v0.3.2
  Downloaded http-body v1.0.1
  Downloaded hexf-parse v0.2.1
  Downloaded rustc-hash v1.1.0
  Downloaded byteorder v1.4.3
  Downloaded bytemuck_derive v1.4.1
  Downloaded cfg-if v1.0.0
  Downloaded bit_field v0.10.2
  Downloaded color_quant v1.1.0
  Downloaded cfg_aliases v0.1.1
  Downloaded bit-set v0.6.0
  Downloaded equivalent v1.0.1
  Downloaded axum-core v0.4.3
  Downloaded adler v1.0.2
  Downloaded gpu-descriptor v0.3.0
  Downloaded foreign-types-shared v0.1.1
  Downloaded gpu-descriptor-types v0.2.0
  Downloaded gpu-alloc-types v0.3.0
  Downloaded arrayref v0.3.7
  Downloaded futures-macro v0.3.30
  Downloaded data-encoding v2.6.0
  Downloaded futures-io v0.3.30
  Downloaded cpufeatures v0.2.13
  Downloaded httpdate v1.0.2
  Downloaded cmake v0.1.50
  Downloaded chunked_transfer v1.4.1
  Downloaded tracing-serde v0.1.3
  Downloaded cexpr v0.6.0
  Downloaded block-buffer v0.10.4
  Downloaded bitflags v1.3.2
  Downloaded tinyvec_macros v0.1.1
  Downloaded sync_wrapper v1.0.1
  Downloaded sync_wrapper v0.1.2
  Downloaded bit-vec v0.7.0
  Downloaded ascii v1.1.0
  Downloaded allocator-api2 v0.2.16
  Downloaded http-body-util v0.1.2
  Downloaded futures-task v0.3.30
  Downloaded form_urlencoded v1.2.0
  Downloaded font-types v0.6.0
  Downloaded ffmpeg-sys-next v7.0.2
  Downloaded etagere v0.2.13
  Downloaded dyn-clone v1.0.14
  Downloaded digest v0.10.7
  Downloaded bitflags v2.6.0
  Downloaded approx v0.5.1
  Downloaded tower-layer v0.3.3
  Downloaded autocfg v1.1.0
  Downloaded atomic-waker v1.1.2
  Downloaded sys-locale v0.3.1
  Downloaded svg_fmt v0.4.1
  Downloaded httparse v1.8.0
  Downloaded rand_core v0.6.4
  Downloaded half v2.2.1
  Downloaded futures-channel v0.3.28
  Downloaded fontconfig-parser v0.5.3
  Downloaded flume v0.10.14
  Downloaded bytes v1.4.0
  Downloaded strict-num v0.1.1
  Downloaded flate2 v1.0.26
  Downloaded xmlwriter v0.1.0
  Downloaded try-lock v0.2.4
  Downloaded fdeflate v0.3.0
  Downloaded data-url v0.2.0
  Downloaded crossbeam-utils v0.8.16
  Downloaded crossbeam-epoch v0.9.15
  Downloaded cc v1.1.10
  Downloaded tower-service v0.3.2
  Downloaded base64 v0.22.1
  Downloaded base64 v0.21.2
  Downloaded codespan-reporting v0.11.1
  Downloaded clang-sys v1.6.1
  Downloaded bytemuck v1.13.1
  Downloaded anyhow v1.0.72
  Downloaded strength_reduce v0.2.4
  Downloaded memoffset v0.6.5
  Downloaded simplecss v0.2.1
  Downloaded ahash v0.8.11
  Downloaded tracing-log v0.2.0
  Downloaded sha1 v0.10.6
  Downloaded fs_extra v1.3.0
  Downloaded crc32fast v1.3.2
  Downloaded async-trait v0.1.81
  Downloaded crossbeam-channel v0.5.8
  Downloaded arrayvec v0.7.4
  Downloaded shlex v1.1.0
  Downloaded self_cell v1.0.4
  Downloaded gpu-alloc v0.6.0
  Downloaded glob v0.3.1
  Downloaded getrandom v0.2.10
  Downloaded generic-array v0.14.7
  Downloaded futures-sink v0.3.30
  Downloaded futures-core v0.3.30
  Downloaded fnv v1.0.7
  Downloaded float-cmp v0.9.0
  Downloaded either v1.9.0
  Downloaded document-features v0.2.10
  Downloaded crypto-common v0.1.6
  Downloaded crossbeam-deque v0.8.3
  Downloaded serde_urlencoded v0.7.1
  Downloaded rustc-hash v2.0.0
  Downloaded shared_memory v0.12.4
  Downloaded lock_api v0.4.10
  Downloaded unicode-ccc v0.1.2
  Downloaded roxmltree v0.18.0
  Downloaded num_cpus v1.16.0
  Downloaded nu-ansi-term v0.46.0
  Downloaded rayon-core v1.11.0
  Downloaded once_cell v1.19.0
  Downloaded untrusted v0.9.0
  Downloaded num-rational v0.4.1
  Downloaded unicode-width v0.1.10
  Downloaded unicode-linebreak v0.1.5
  Downloaded unicode-bidi-mirroring v0.2.0
  Downloaded unicode-bidi-mirroring v0.1.0
  Downloaded transpose v0.2.3
  Downloaded thiserror v1.0.63
  Downloaded subtle v2.6.1
  Downloaded http v1.1.0
  Downloaded hashbrown v0.14.0
  Downloaded fontdb v0.16.2
  Downloaded euclid v0.22.9
  Downloaded num-traits v0.2.16
  Downloaded rawpointer v0.2.1
  Downloaded tokio-macros v2.3.0
  Downloaded hashbrown v0.12.3
  Downloaded ffmpeg-next v7.0.4
  Downloaded khronos-egl v6.0.0
  Downloaded xmlparser v0.13.5
  Downloaded serde_path_to_error v0.1.16
  Downloaded miniz_oxide v0.7.1
  Downloaded siphasher v0.3.10
  Downloaded rustversion v1.0.17
  Downloaded lazycell v1.3.0
  Downloaded axum v0.7.5
  Downloaded simd-adler32 v0.3.7
  Downloaded lazy_static v1.4.0
  Downloaded rand_chacha v0.3.1
  Downloaded lebe v0.5.2
  Downloaded realfft v3.3.0
  Downloaded rctree v0.5.0
  Downloaded static_assertions v1.1.0
  Downloaded aho-corasick v1.0.2
  Downloaded scopeguard v1.2.0
  Downloaded libloading v0.8.0
  Downloaded libloading v0.7.4
  Downloaded raw-window-handle v0.6.2
  Downloaded mio v0.8.11
  Downloaded h2 v0.4.6
  Downloaded termcolor v1.4.1
  Downloaded signal-hook-registry v1.4.1
  Downloaded rustls-pemfile v2.1.3
  Downloaded pollster v0.3.0
  Downloaded regex-automata v0.1.10
  Downloaded glow v0.13.1
  Downloaded unicode-xid v0.2.4
  Downloaded unicode-ccc v0.2.0
  Downloaded thiserror-impl v1.0.63
  Downloaded hyper v1.4.1
  Downloaded futures-util v0.3.30
  Downloaded tokio-native-tls v0.3.1
  Downloaded matchers v0.1.0
  Downloaded rangemap v1.5.1
  Downloaded imagesize v0.12.0
  Downloaded primal-check v0.3.4
  Downloaded unicode-vo v0.1.0
  Downloaded pin-utils v0.1.0
  Downloaded memoffset v0.7.1
  Downloaded mime v0.3.17
  Downloaded pico-args v0.5.0
  Downloaded zeroize v1.8.1
  Downloaded memoffset v0.9.0
  Downloaded pkg-config v0.3.27
  Downloaded lru v0.12.4
  Downloaded svgtypes v0.11.0
  Downloaded rgb v0.8.36
  Downloaded memmap2 v0.6.2
  Downloaded paste v1.0.14
  Downloaded matchit v0.7.3
  Downloaded percent-encoding v2.3.0
  Downloaded ppv-lite86 v0.2.17
  Downloaded parking_lot_core v0.9.8
  Downloaded overload v0.1.1
  Downloaded thread_local v1.1.8
  Downloaded rand v0.8.5
  Downloaded quote v1.0.36
  Downloaded spirv v0.3.0+sdk-1.3.268.0
  Downloaded profiling v1.0.9
  Downloaded tokio-rustls v0.26.0
  Downloaded unicode-general-category v0.6.0
  Downloaded openssl-probe v0.1.5
  Downloaded openssl-macros v0.1.1
  Downloaded memmap2 v0.9.4
  Downloaded num-complex v0.4.4
  Downloaded indexmap v1.9.3
  Downloaded serde_derive_internals v0.26.0
  Downloaded hyper-rustls v0.27.2
  Downloaded want v0.3.1
  Downloaded kurbo v0.9.5
  Downloaded num-integer v0.1.45
  Downloaded parking_lot v0.12.1
  Downloaded usvg-text-layout v0.35.0
  Downloaded exr v1.7.0
  Downloaded proc-macro2 v1.0.86
  Downloaded opus v0.3.0
  Downloaded matrixmultiply v0.3.7
  Downloaded slab v0.4.8
  Downloaded native-tls v0.2.11
  Downloaded signal-hook v0.3.17
  Downloaded version_check v0.9.4
  Downloaded qoi v0.4.1
  Downloaded indexmap v2.0.1
  Downloaded spin v0.9.8
  Downloaded smallvec v1.13.2
  Downloaded rustls-pki-types v1.8.0
  Downloaded libm v0.2.8
  Downloaded jobserver v0.1.32
  Downloaded tracing-attributes v0.1.27
  Downloaded bindgen v0.69.4
  Downloaded rubato v0.15.0
  Downloaded tokio-tungstenite v0.21.0
  Downloaded minimal-lexical v0.2.1
  Downloaded log v0.4.22
  Downloaded litrs v0.4.1
  Downloaded png v0.17.9
  Downloaded memchr v2.7.4
  Downloaded num-bigint v0.4.4
  Downloaded ryu v1.0.15
  Downloaded rayon v1.7.0
  Downloaded socket2 v0.5.7
  Downloaded tiny_http v0.12.0
  Downloaded typenum v1.16.0
  Downloaded nom v7.1.3
  Downloaded image v0.24.7
  Downloaded nanorand v0.7.0
  Downloaded wgpu v22.1.0
  Downloaded itertools v0.12.1
  Downloaded tiny-skia-path v0.10.0
  Downloaded fontdb v0.14.1
  Downloaded usvg-tree v0.35.0
  Downloaded zerocopy v0.7.35
  Downloaded hyper-tls v0.6.0
  Downloaded idna v0.4.0
  Downloaded serde_json v1.0.104
  Downloaded openssl v0.10.55
  Downloaded unicode-bidi v0.3.13
  Downloaded regex v1.10.6
  Downloaded tinyvec v1.6.0
  Downloaded unicode-script v0.5.5
  Downloaded unicode-properties v0.1.2
  Downloaded unicode-ident v1.0.11
  Downloaded pin-project-lite v0.2.14
  Downloaded slotmap v1.0.6
  Downloaded serde_derive v1.0.208
  Downloaded itoa v1.0.9
  Downloaded utf-8 v0.7.6
  Downloaded renderdoc-sys v1.1.0
  Downloaded pin-project-internal v1.1.2
  Downloaded regex-syntax v0.6.29
  Downloaded simba v0.8.1
  Downloaded ash v0.38.0+1.3.281
  Downloaded read-fonts v0.20.0
  Downloaded nix v0.23.2
  Downloaded safe_arch v0.7.1
  Downloaded nalgebra-glm v0.18.0
  Downloaded nix v0.26.4
  Downloaded zune-inflate v0.2.54
  Downloaded usvg v0.35.0
  Downloaded ipnet v2.8.0
  Downloaded zeno v0.2.2
  Downloaded yazi v0.1.6
  Downloaded tungstenite v0.21.0
  Downloaded regex-syntax v0.8.4
  Downloaded tracing-core v0.1.32
  Downloaded sharded-slab v0.1.7
  Downloaded tracing v0.1.40
  Downloaded rtcp v0.10.1
  Downloaded regex-automata v0.4.7
  Downloaded gif v0.12.0
  Downloaded weezl v0.1.7
  Downloaded openssl-sys v0.9.90
  Downloaded unicode-segmentation v1.10.1
  Downloaded serde v1.0.208
  Downloaded tokio-util v0.7.8
  Downloaded hyper-util v0.1.7
  Downloaded naga v22.1.0
  Downloaded jpeg-decoder v0.3.0
  Downloaded rtp v0.9.0
  Downloaded pin-project v1.1.2
  Downloaded usvg-parser v0.35.0
  Downloaded wgpu-types v22.0.0
  Downloaded libc v0.2.158
  Downloaded resvg v0.35.0
  Downloaded audiopus_sys v0.2.2
  Downloaded tower v0.4.13
  Downloaded url v2.4.1
  Downloaded unicode-normalization v0.1.22
  Downloaded wide v0.7.11
  Downloaded encoding_rs v0.8.32
  Downloaded webrtc-util v0.8.1
  Downloaded ttf-parser v0.19.1
  Downloaded cosmic-text v0.12.1
  Downloaded ttf-parser v0.20.0
  Downloaded ttf-parser v0.18.1
  Downloaded fdk-aac-sys v0.5.0
  Downloaded ttf-parser v0.21.1
  Downloaded tracing-subscriber v0.3.18
  Downloaded tiny-skia v0.10.0
  Downloaded skrifa v0.20.0
  Downloaded rustls-webpki v0.102.6
  Downloaded reqwest v0.12.7
  Downloaded rustybuzz v0.7.0
  Downloaded ring v0.17.8
  Downloaded syn v1.0.109
  Downloaded swash v0.1.18
  Downloaded rustybuzz v0.14.1
  Downloaded wgpu-core v22.1.0
  Downloaded syn v2.0.75
  Downloaded rustfft v6.2.0
  Downloaded wgpu-hal v22.0.0
  Downloaded nalgebra v0.32.3
  Downloaded rustls v0.23.12
  Downloaded tokio v1.38.1
  Downloaded tiff v0.9.0
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.11
   Compiling cfg-if v1.0.0
   Compiling libc v0.2.158
   Compiling quote v1.0.36
   Compiling syn v2.0.75
   Compiling version_check v0.9.4
   Compiling log v0.4.22
   Compiling scopeguard v1.2.0
   Compiling once_cell v1.19.0
   Compiling smallvec v1.13.2
   Compiling lock_api v0.4.10
   Compiling jobserver v0.1.32
   Compiling cc v1.1.10
   Compiling num_cpus v1.16.0
   Compiling bytemuck_derive v1.4.1
   Compiling bytemuck v1.13.1
   Compiling parking_lot_core v0.9.8
   Compiling ahash v0.8.11
   Compiling num-traits v0.2.16
   Compiling bytes v1.4.0
   Compiling pin-project-lite v0.2.14
   Compiling pkg-config v0.3.27
   Compiling zerocopy v0.7.35
   Compiling parking_lot v0.12.1
   Compiling bitflags v1.3.2
   Compiling itoa v1.0.9
   Compiling allocator-api2 v0.2.16
   Compiling futures-core v0.3.30
   Compiling serde v1.0.208
   Compiling hashbrown v0.14.0
   Compiling serde_derive v1.0.208
   Compiling futures-sink v0.3.30
   Compiling equivalent v1.0.1
   Compiling indexmap v2.0.1
   Compiling signal-hook-registry v1.4.1
   Compiling tinyvec_macros v0.1.1
   Compiling tinyvec v1.6.0
   Compiling tokio-macros v2.3.0
   Compiling getrandom v0.2.10
   Compiling mio v0.8.11
   Compiling socket2 v0.5.7
   Compiling pin-utils v0.1.0
   Compiling thiserror v1.0.63
   Compiling bitflags v2.6.0
   Compiling crossbeam-utils v0.8.16
   Compiling tokio v1.38.1
   Compiling thiserror-impl v1.0.63
   Compiling tracing-core v0.1.32
   Compiling num-integer v0.1.45
   Compiling memoffset v0.9.0
   Compiling slab v0.4.8
   Compiling arrayvec v0.7.4
   Compiling cfg_aliases v0.1.1
   Compiling crossbeam-epoch v0.9.15
   Compiling fnv v1.0.7
   Compiling simd-adler32 v0.3.7
   Compiling typenum v1.16.0
   Compiling http v1.1.0
   Compiling tracing-attributes v0.1.27
   Compiling unicode-bidi v0.3.13
   Compiling tracing v0.1.40
   Compiling futures-macro v0.3.30
   Compiling futures-task v0.3.30
   Compiling futures-io v0.3.30
   Compiling adler v1.0.2
   Compiling memchr v2.7.4
   Compiling rayon-core v1.11.0
   Compiling xmlparser v0.13.5
   Compiling crc32fast v1.3.2
   Compiling roxmltree v0.18.0
   Compiling futures-util v0.3.30
   Compiling miniz_oxide v0.7.1
   Compiling crossbeam-deque v0.8.3
   Compiling crossbeam-channel v0.5.8
   Compiling rand_core v0.6.4
   Compiling openssl-sys v0.9.90
   Compiling pin-project-internal v1.1.2
   Compiling slotmap v1.0.6
   Compiling glob v0.3.1
   Compiling ppv-lite86 v0.2.17
   Compiling httparse v1.8.0
   Compiling rand_chacha v0.3.1
   Compiling pin-project v1.1.2
   Compiling clang-sys v1.6.1
   Compiling float-cmp v0.9.0
   Compiling strict-num v0.1.1
   Compiling flate2 v1.0.26
   Compiling rand v0.8.5
   Compiling fontconfig-parser v0.5.3
   Compiling http-body v1.0.1
   Compiling generic-array v0.14.7
   Compiling num-bigint v0.4.4
   Compiling arrayref v0.3.7
   Compiling either v1.9.0
   Compiling ryu v1.0.15
   Compiling serde_json v1.0.104
   Compiling percent-encoding v2.3.0
   Compiling futures-channel v0.3.28
   Compiling minimal-lexical v0.2.1
   Compiling unicode-script v0.5.5
   Compiling regex-syntax v0.8.4
   Compiling nom v7.1.3
   Compiling regex-automata v0.4.7
   Compiling form_urlencoded v1.2.0
   Compiling rayon v1.7.0
   Compiling libloading v0.7.4
   Compiling tiny-skia-path v0.10.0
   Compiling tokio-util v0.7.8
   Compiling naga v22.1.0
   Compiling kurbo v0.9.5
   Compiling unicode-normalization v0.1.22
   Compiling num-complex v0.4.4
   Compiling libloading v0.8.0
   Compiling num-rational v0.4.1
   Compiling rustversion v1.0.17
   Compiling ash v0.38.0+1.3.281
   Compiling foreign-types-shared v0.1.1
   Compiling atomic-waker v1.1.2
   Compiling bindgen v0.69.4
   Compiling tower-layer v0.3.3
   Compiling termcolor v1.4.1
   Compiling siphasher v0.3.10
   Compiling try-lock v0.2.4
   Compiling rustc-hash v1.1.0
   Compiling khronos-egl v6.0.0
   Compiling bit-vec v0.7.0
   Compiling httpdate v1.0.2
   Compiling tower-service v0.3.2
   Compiling openssl v0.10.55
   Compiling unicode-general-category v0.6.0
   Compiling unicode-width v0.1.10
   Compiling codespan-reporting v0.11.1
   Compiling itertools v0.12.1
   Compiling bit-set v0.6.0
   Compiling svgtypes v0.11.0
   Compiling want v0.3.1
   Compiling h2 v0.4.6
   Compiling foreign-types v0.3.2
   Compiling idna v0.4.0
   Compiling regex v1.10.6
   Compiling cexpr v0.6.0
   Compiling wgpu-hal v22.0.0
   Compiling gpu-alloc-types v0.3.0
   Compiling spirv v0.3.0+sdk-1.3.268.0
   Compiling gpu-descriptor-types v0.2.0
   Compiling font-types v0.6.0
   Compiling openssl-macros v0.1.1
   Compiling memoffset v0.7.1
   Compiling lazy_static v1.4.0
   Compiling unicode-xid v0.2.4
   Compiling libm v0.2.8
   Compiling rctree v0.5.0
   Compiling byteorder v1.4.3
   Compiling lazycell v1.3.0
   Compiling native-tls v0.2.11
   Compiling hexf-parse v0.2.1
   Compiling shlex v1.1.0
   Compiling paste v1.0.14
   Compiling usvg-tree v0.35.0
   Compiling read-fonts v0.20.0
error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in trait method return types
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:69:28
   |
69 |     fn ordered_values() -> impl DoubleEndedIterator<Item = u32>;
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in trait method return types
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:75:58
   |
75 |     fn ordered_values_range(range: RangeInclusive<T>) -> impl DoubleEndedIterator<Item = u32>;
   |                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:777:28
    |
777 |     fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:781:60
    |
781 |     fn ordered_values_range(range: RangeInclusive<u32>) -> impl DoubleEndedIterator<Item = u32> {
    |                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:803:28
    |
803 |     fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:807:60
    |
807 |     fn ordered_values_range(range: RangeInclusive<u16>) -> impl DoubleEndedIterator<Item = u32> {
    |                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:829:28
    |
829 |     fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:833:59
    |
833 |     fn ordered_values_range(range: RangeInclusive<u8>) -> impl DoubleEndedIterator<Item = u32> {
    |                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:855:28
    |
855 |     fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:861:10
    |
861 |     ) -> impl DoubleEndedIterator<Item = u32> {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:883:28
    |
883 |     fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in `impl` method return types
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/read-fonts-0.20.0/src/collections/int_set/mod.rs:889:10
    |
889 |     ) -> impl DoubleEndedIterator<Item = u32> {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #91611 <[https://github.com/rust-lang/rust/issues/91611>](https://github.com/rust-lang/rust/issues/91611%3E) for more information

For more information about this error, try `rustc --explain E0562`.
error: could not compile `read-fonts` (lib) due to 12 previous errors
warning: build failed, waiting for other jobs to finish...
The command '/bin/bash -o pipefail -c source ~/.cargo/env && cargo build --release --no-default-features' returned a non-zero code: 101
```

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -1,5 +1,8 @@
 name: docker
 on:
+    workflow_dispatch: {}
+    schedule:
+      - cron: '0 0 * * 0' # once a week
     push:
         branches: [master]
         paths:

--- a/build_tools/docker/full.Dockerfile
+++ b/build_tools/docker/full.Dockerfile
@@ -4,7 +4,7 @@ FROM ubuntu:noble-20240423 as builder
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 ARG USERNAME=compositor
-ARG RUST_VERSION=1.74
+ARG RUST_VERSION=1.81
 
 ENV DEBIAN_FRONTEND=noninteractive
 
@@ -25,7 +25,7 @@ WORKDIR /root/project
 RUN source ~/.cargo/env && cargo build --release
 
 # Runtime image
-FROM ubuntu:mantic-20231011 
+FROM ubuntu:noble-20240423
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 

--- a/build_tools/docker/slim.Dockerfile
+++ b/build_tools/docker/slim.Dockerfile
@@ -4,7 +4,7 @@ FROM ubuntu:noble-20240423 as builder
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 ARG USERNAME=compositor
-ARG RUST_VERSION=1.74
+ARG RUST_VERSION=1.81
 
 ENV DEBIAN_FRONTEND=noninteractive
 
@@ -24,7 +24,7 @@ WORKDIR /root/project
 RUN source ~/.cargo/env && cargo build --release --no-default-features
 
 # Runtime image
-FROM ubuntu:mantic-20231011
+FROM ubuntu:noble-20240423
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 

